devimint/src/federation.rs

```diff
@@ -764,6 +766,80 @@ impl Federation {
         Ok(())
     }
 
+    pub async fn pegout_gateways(
+        &self,
+        amount: u64,
+        gateways: Vec<&super::gatewayd::Gatewayd>,
+    ) -> Result<()> {
+        info!(amount, "Pegging-out gateway funds");
+        let fed_id = self.calculate_federation_id();
+        let mut peg_outs: BTreeMap<LightningNodeType, (Amount, WithdrawResponse)> = BTreeMap::new();
+        for gw in gateways.clone() {
+            let prev_fed_ecash_balance = gw
+                .get_balances()
+                .await?
+                .ecash_balances
+                .into_iter()
+                .find(|fed| fed.federation_id.to_string() == fed_id)
+                .expect("Gateway has not joined federation")
+                .ecash_balance_msats;
+
+            let pegout_address = self.bitcoind.get_new_address().await?;
+            let value = cmd!(
+                gw,
+                "withdraw",
+                "--federation-id",
+                fed_id,
+                "--amount",
+                amount,
+                "--address",
+                pegout_address
+            )
+            .out_json()
+            .await?;
+            let response: WithdrawResponse = serde_json::from_value(value)?;
+            peg_outs.insert(gw.ln_type(), (prev_fed_ecash_balance, response));
+        }
+        self.bitcoind.mine_blocks(21).await?;
+
+        try_join_all(
+            peg_outs
+                .iter()
+                .map(|res| self.bitcoind.poll_get_transaction(res.1 .1.txid)),
+        )
+        .await?;
```

Comment:
Fixed

Reply:
Looks like you're using a slightly more verbose version of what I have above. Can you use `.values()` instead of `.iter()` and change `(_, (_, res))` to `(_, res)`?

Reply:
ah yes, thanks for checking. Fixed